docs/permission/plugin-authors/authorize-the-update-endpoint.md

```diff
@@ -0,0 +1,160 @@
+
+Before diving into the extra steps needed for supporting conditional decisions, let's go back to the permission policy's handle function used by your adopters and try to authorize our new permission.
+
+Let's edit `packages/backend/src/plugins/permission.ts`
+
+```diff
+    if (request.permission.name === 'todos.list.create') {
+      return {
+        result: AuthorizeResult.DENY,
+      };
+    }
++   if (request.permission.resourceType === 'todo-item') {
++     if (request.permission.attributes.action === 'update') {
++       return {
++         result: AuthorizeResult.CONDITIONAL,
++         pluginId: 'todolist',
++         resourceType: 'todo-item',
++         conditions: {
++           rule: 'IS_OWNER',
++           params: [user?.identity.userEntityRef],
++         },
++       };
++     }
++   }
+    return {
+      result: AuthorizeResult.ALLOW,
+    };
+```
+
+This is what happens when a _Conditional Decision_ is returned. We are saying:
+
+> Hey permission framework, I can't make a decision alone. Please go to the plugin with id `todolist` and ask it to apply these conditions.
+
+Now if we try to edit an item from the UI, we should spot the following error in the backend's console:
+
+```
+backstage error Unexpected response from plugin upstream when applying conditions.
+Expected 200 but got 404 - Not Found type=errorHandler stack=Error:
+Unexpected response from plugin upstream when applying conditions. Expected 200 but got 404 - Not Found
+```
+
+This happens because our plugin should have exposed a specific endpoint, used by the permission framework to apply conditional decisions. The new endpoint should also be able to support some conditions. In our case, `IS_OWNER` is the only type of condition we want to support. You can add as many built-in conditions as you like to your plugin, and you can also allow Backstage integrators to supply more conditions when starting your backend if you want.
+
+Install the missing module:
+
+```
+$ yarn workspace @internal/plugin-todo-list-backend add @backstage/plugin-permission-node
+```
+
+Create a new `plugins/todo-list-backend/src/service/rules.ts` file and append the following code:
+
+```diff
++ import { createPermissionIntegrationRouter } from '@backstage/plugin-permission-node';
++ import { makeCreatePermissionRule } from '@backstage/plugin-permission-node';
++ import { Todo, TodoFilter } from './todos';
+
++ const createTodoListPermissionRule = makeCreatePermissionRule<
++   Todo,
++   TodoFilter
++ >();
+
++ export const isOwner = createTodoListPermissionRule({
++   name: 'IS_OWNER',
++   description: 'Should allow only if the todo belongs to the user',
++   apply: (resource, userId) => {
++     return resource.author === userId;
++   },
++   toQuery: userId => {
++     return resource => resource.author === userId;
++   },
++ });
+
++ export const rules = [isOwner];
```

Comment:
Looks like an old comment got lost here:

> This should probably be an object for convenience when calling createConditionExports.

```suggestion
+ export const rules = { isOwner };
```

Reply:
yeah, I've exported as an array for convenience when passing the rules to `createPermissionIntegrationRouter` 😅

Makes sense to export as object since it will make things easier for plugin adopters